src/MiniCover/Commands/Options/MiniCoverOption.cs

```diff
@@ -15,18 +17,24 @@ public TValue Value
         {
             get
             {
-                if (Validated) return ValueField;
-                throw new MemberAccessException("Option should be invalidate before Value access");
+                if (!validated)
+                    throw new MemberAccessException("Option should be validated before Value access");
+
+                return value;
             }
         }
 
-        protected CommandOption Option { get; set; }
+        public virtual void AddTo(CommandLineApplication command)
+        {
+            option = command.Option(OptionTemplate, Description, Type);
+        }
 
-        public virtual void Initialize(CommandLineApplication commandContext)
+        public void Validate()
         {
-            Option = commandContext.Option(OptionTemplate, Description, Type);
+            value = GetOptionValue(option);
+            validated = true;
```

Comment:
Moved some validate logic to the base class also. Setting validated to true makes sense in this class because line 18 checks for validated variable. It can even be privated.